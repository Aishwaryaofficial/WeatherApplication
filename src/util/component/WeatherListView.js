import React, { useState, useEffect } from "react";
import {
    StyleSheet,
    Text,
    View,
    Image,
    SafeAreaView,
    SectionList,
    ActivityIndicator,
    TouchableOpacity
} from "react-native";
import AppStyle from '../AppStyle';
import AppConstant from '../AppConstant';
import moment from "moment";

const WeatherListView = (props) => {
    //console.log(props.wData);
    const Item = ({ item }) => (
        <View style={AppStyle.itemContainer}>
            <TouchableOpacity style={{flexDirection: 'row' ,width:'100%'}} onPress={() => {
                props.goToWeatherDetail(item);
            }}>
                <Image
                    style={{ width: 80, height: 80, resizeMode: 'cover' }}
                    source={{ uri: 'http://openweathermap.org/img/wn/' + item.weather[0].icon + '@2x.png' }}
                />
                <View style={AppStyle.item}>
                    <Text style={{ fontSize: 20, textAlign: 'left' }}>{moment(item.dt_txt, 'YYYY-MM-DD h:mm:ss').format('h:mm a')}</Text>
                    <Text style={AppStyle.temp}>{parseFloat(item.main.temp - 273.15).toFixed(2)}Â°C</Text>
                </View>
            </TouchableOpacity>
        </View>
    );
   
    if (props.wData == null) {
        return (
            <SafeAreaView>
                <View style={{ flex: 1, padding: 20  }}>
                    <ActivityIndicator />
                </View>
            </SafeAreaView>
        )
    }
    return (
        <>
            <SafeAreaView style={AppStyle.weatherListContainer}>
                <SectionList
                    sections={props.wData}
                    keyExtractor={(item) => item.dt_txt}
                    renderItem={({ item }) => <Item item={item} />}
                    renderSectionHeader={({ section }) => (
                        <View style={AppStyle.header}>
                            <Text style={AppStyle.title}>
                                {moment(section.key, 'YYYY-MM-DD h:mm:ss').format('MMMM Do YYYY')}
                            </Text>
                        </View>)
                    }
                />
            </SafeAreaView>
        </>
    );
};

export default WeatherListView;
