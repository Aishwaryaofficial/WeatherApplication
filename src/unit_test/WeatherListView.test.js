import {configure , shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import WeatherListView from '../util/component/WeatherListView';
import { SectionList } from 'react-native';
import React from 'react';


const weatherArr = [
   {
    "key":"2020-05-11",
    "data":{"dt":1589220000,"main":{"temp":305.15,"feels_like":301.8,"temp_min":305.15,"temp_max":306.2,"pressure":1007,"sea_level":1007,"grnd_level":983,"humidity":37,"temp_kf":-1.05},"weather":[{"id":500,"main":"Rain","description":"light rain","icon":"10n"}],"clouds":{"all":55},"wind":{"speed":7.34,"deg":314},"rain":{"3h":0.32},"sys":{"pod":"n"},"dt_txt":"2020-05-11 18:00:00"}
   },{
    "key":"2020-05-11",
    "data":{"dt":1589220000,"main":{"temp":305.15,"feels_like":301.8,"temp_min":305.15,"temp_max":306.2,"pressure":1007,"sea_level":1007,"grnd_level":983,"humidity":37,"temp_kf":-1.05},"weather":[{"id":500,"main":"Rain","description":"light rain","icon":"10n"}],"clouds":{"all":55},"wind":{"speed":7.34,"deg":314},"rain":{"3h":0.32},"sys":{"pod":"n"},"dt_txt":"2020-05-11 18:00:00"}
   }
]
configure({adapter: new Adapter()});
describe('Testing DataListSection component', () => {
  it('renders as expected', () => {
    const dataList = shallow(
      <WeatherListView weatherData = {weatherArr} />
    );
    expect(dataList).toMatchSnapshot();
    dataList.setProps({ weatherData: [weatherArr] });
    expect(dataList).toMatchSnapshot();
    const flatList = dataList.find('SectionList');
    const item = flatList.renderProp('renderItem')({ item: weatherArr });
    const item1 = flatList.renderProp('renderSectionHeader')
    const item3 = flatList.renderProp('keyExtractor')({ item: weatherArr });
    
    expect(item3).toMatchSnapshot();
    expect(item).toMatchSnapshot();
    expect(item1).toMatchSnapshot();


  });
});