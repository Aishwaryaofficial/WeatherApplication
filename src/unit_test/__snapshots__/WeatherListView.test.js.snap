// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing DataListSection component renders as expected 1`] = `
<Fragment>
  <ForwardRef(SafeAreaView)
    style={
      Object {
        "backgroundColor": "black",
        "flex": 1,
        "marginTop": 0,
        "width": "100%",
      }
    }
  >
    <SectionList
      data={Array []}
      disableVirtualization={false}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      onEndReachedThreshold={2}
      renderItem={[Function]}
      renderSectionHeader={[Function]}
      scrollEventThrottle={50}
      stickySectionHeadersEnabled={true}
      updateCellsBatchingPeriod={50}
      windowSize={21}
    />
  </ForwardRef(SafeAreaView)>
</Fragment>
`;

exports[`Testing DataListSection component renders as expected 2`] = `
<Fragment>
  <ForwardRef(SafeAreaView)
    style={
      Object {
        "backgroundColor": "black",
        "flex": 1,
        "marginTop": 0,
        "width": "100%",
      }
    }
  >
    <SectionList
      data={Array []}
      disableVirtualization={false}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      onEndReachedThreshold={2}
      renderItem={[Function]}
      renderSectionHeader={[Function]}
      scrollEventThrottle={50}
      stickySectionHeadersEnabled={true}
      updateCellsBatchingPeriod={50}
      windowSize={21}
    />
  </ForwardRef(SafeAreaView)>
</Fragment>
`;

exports[`Testing DataListSection component renders as expected 3`] = `""`;

exports[`Testing DataListSection component renders as expected 4`] = `
<Item
  item={
    Array [
      Object {
        "data": Object {
          "clouds": Object {
            "all": 55,
          },
          "dt": 1589220000,
          "dt_txt": "2020-05-11 18:00:00",
          "main": Object {
            "feels_like": 301.8,
            "grnd_level": 983,
            "humidity": 37,
            "pressure": 1007,
            "sea_level": 1007,
            "temp": 305.15,
            "temp_kf": -1.05,
            "temp_max": 306.2,
            "temp_min": 305.15,
          },
          "rain": Object {
            "3h": 0.32,
          },
          "sys": Object {
            "pod": "n",
          },
          "weather": Array [
            Object {
              "description": "light rain",
              "icon": "10n",
              "id": 500,
              "main": "Rain",
            },
          ],
          "wind": Object {
            "deg": 314,
            "speed": 7.34,
          },
        },
        "key": "2020-05-11",
      },
      Object {
        "data": Object {
          "clouds": Object {
            "all": 55,
          },
          "dt": 1589220000,
          "dt_txt": "2020-05-11 18:00:00",
          "main": Object {
            "feels_like": 301.8,
            "grnd_level": 983,
            "humidity": 37,
            "pressure": 1007,
            "sea_level": 1007,
            "temp": 305.15,
            "temp_kf": -1.05,
            "temp_max": 306.2,
            "temp_min": 305.15,
          },
          "rain": Object {
            "3h": 0.32,
          },
          "sys": Object {
            "pod": "n",
          },
          "weather": Array [
            Object {
              "description": "light rain",
              "icon": "10n",
              "id": 500,
              "main": "Rain",
            },
          ],
          "wind": Object {
            "deg": 314,
            "speed": 7.34,
          },
        },
        "key": "2020-05-11",
      },
    ]
  }
/>
`;

exports[`Testing DataListSection component renders as expected 5`] = `[Function]`;
